qiskit/circuit/quantumcircuit.py

```diff
@@ -1400,24 +1401,47 @@ def _increment_instances(cls):
         cls.instances += 1
 
     @classmethod
+    @deprecate_func(
+        since=1.2,
+        removal_timeline="in qiskit 2.0",
+        additional_msg="This method is only used as an internal helper "
+        "and will be removed with no replacement.",
+    )
     def cls_instances(cls) -> int:
         """Return the current number of instances of this class,
         useful for auto naming."""
         return cls.instances
 
     @classmethod
+    def _cls_instances(cls) -> int:
+        """Return the current number of instances of this class,
+        useful for auto naming."""
+        return cls.instances
+
+    @classmethod
+    @deprecate_func(
+        since=1.2,
+        removal_timeline="in qiskit 2.0",
```

Comment:
Should we uppercase this? 🙂 
```suggestion
        removal_timeline="in Qiskit 2.0",
```

Reply:
I wrote it in lowercase because the beginning of the message writes `qiskit` in lowercase. But I also don't love it. Anyways, I modified the timeline message in [a7b7361](https://github.com/Qiskit/qiskit/pull/12785/commits/a7b73616bc3004b31218fa5d48e23c81845f89cb) to say 
"in the 2.0 release" to be consistent with other deprecations (qobj and assemble).

Reply:
Oh I didn't notice... that seems odd. Maybe we should change that as well 🤔 (not here ofc)